### PR TITLE
Check if PHP is available before trying to run Grumphp (wip)

### PR DESCRIPTION
**Pull request by drupol**
Hi,

Do you think such minor update in the hooks are justified in Grumphp?

Thanks

| Q             | A
| ------------- | ---
| Branch        | master
| Bug fix?      | no
| New feature?  | yes
| BC breaks?    | no
| Deprecations? | no
| Documented?   | no

# New Task Checklist:

- [ ] Are the dependencies added to the composer.json suggestions?
- [ ] Is the doc/tasks.md file updated?
- [ ] Are the task parameters documented?
- [ ] Is the task registered in the tasks.yml file?
- [ ] Does the task contains phpunit tests?
- [ ] Is the configuration having logical allowed types?
- [ ] Does the task run in the correct context?
- [ ] Is the `run()` method readable?
- [ ] Is the `run()` method using the configuration correctly?
- [ ] Are all CI services returning green?

### CODE REVIEW

File: resources/hooks/local/commit-msg [hunk 1]
```diff
@@ -1,5 +1,11 @@
 #!/bin/sh
 
+if ! command -v $(EXEC_GRUMPHP_COMMAND) &> /dev/null
```
Comment (veewee) [inline resources/hooks/local/commit-msg:3]: The exec command can also be a command with cli options and attributes.
For example:

```
grumphp:
    git_hook_variables:
        EXEC_GRUMPHP_COMMAND: '/usr/local/bin/php72'
        EXEC_GRUMPHP_COMMAND: 'lando php'
        EXEC_GRUMPHP_COMMAND: 'fin exec php'
        EXEC_GRUMPHP_COMMAND: ['php', '-c /custom/config.ini']
        EXEC_GRUMPHP_COMMAND: ['docker-compose', 'run', '-T', '--rm', '--no-deps', 'php']
        EXEC_GRUMPHP_COMMAND: 'docker run --rm -it -v $(pwd):/grumphp -w /grumphp webdevops/php:alpine'
        EXEC_GRUMPHP_COMMAND: 'ddev exec php'
```

This won't work in that situation

**Comment by veewee** [comment]
Why would you want to skip the hooks in that case?
Doesn't it sound more reasonable to fail so that the root cause can be fixed by the owner of the machine instead?

I don't see a big benefit of adding this check at this moment.  In what situation would you use it?

**Comment by drupol** [comment]
Right, I understand that sometimes this is not really feasible using `command` in bash, I don't see any better option right now.

> I don't see a big benefit of adding this check at this moment. In what situation would you use it?

**Context:** I'm using Nix locally for PHP development.
**Issue:** Sometimes, I need to make a change in a file without entering an environment containing PHP and all my dependencies (`nix shell github:loophp/nix-shell#env-php82-nts --impure`).
When I do that, `git` refuses to commit stuff because the hook is failing, because PHP doesn't exist. This is precisely the reason why I wanted to make this PR.

**Comment by veewee** [comment]
I'dd suggest to either

1. Commit with `-n` flag in those cases
2. Create your own git hook templates

I don't think this is something we should add to this repo nevertheless.

**Comment by drupol** [comment]
You're right, thanks for the enlightenment.

Have a good day !

**Pull request closed without merging**